clang-tools-extra/docs/clang-tidy/checks/misc/must-use.rst

```diff
@@ -0,0 +1,28 @@
+.. title:: clang-tidy - misc-must-use
+
+misc-must-use
+=============
+
+Allow strictly enforcing that variables are used for specific classes,
+even with they would not be normally warned using `-Wunused-variable` due 
```

Comment:
consider changing check in a way, so it would not overlap with -Wunused-variable

Reply:
Done.